AboutWindow using BAboutWindow

bach5000 opened the pull request:
Should fix #11.
List of changes:
* Deleted the old AboutWindow.cpp, .h, and other unnecessary files
* Changed Makefile to support BAboutWindow

The "About Slayer" window should be fine now, except for copyright info and the size of the window.
Any suggestions/tips are welcome :)

humdingerb commented on src/MainWindow.cpp:
```diff
@@ -12,6 +12,8 @@
 #include "SettingsWindow.h"
 #include "SlayerApp.h"
 #include "AboutWindow.h"
```
Remove #include "AboutWindow.h"

humdingerb commented on src/MainWindow.cpp:
```diff
@@ -157,13 +159,17 @@ void MainWindow::MessageReceived(BMessage *message)
-				about->Activate(true);
-				about->Unlock();
-			}
+			BAboutWindow* fAboutWin = new BAboutWindow(B_TRANSLATE_SYSTEM_NAME("Slayer"), SLAYER_VERSION);
```
The 2nd parameter is the apps signature, not SLAYER_VERSION.
Would be good to keep the signature one constant that'd be used throughout the code.

bach5000 replied:
@humdingerb What would be Slayer's signature? (actually a better question is "what even is a signature?")

humdingerb replied:
It's a unique identifier used when creating the BApplication object, see the BeBook on BApplication. It's used by the system e.g. to send BMessages to the correct app.

bach5000 replied:
@humdingerb I'm still lost on the signature thing. The BeBook seems to use it as some sort of a directory.

humdingerb replied:
Slayer's signature is used here: https://github.com/HaikuArchives/Slayer/blob/master/src/SlayerApp.cpp#L27
Best used as a constant, not copied at several places.

bach5000 replied:
@humdingerb So that means I would use "slayer_signature", since it's defined as a constant for the signature. Thanks, @humdingerb :)

humdingerb commented on src/Makefile:
```diff
@@ -73,7 +74,7 @@ RSRCS=
 #		and it's name
 #		library: my_lib.a entry: my_lib.a or path/my_lib.a
-LIBS= be translation shared
+LIBS= be translation localestub
```
Needs shared for string_for_size

humdingerb commented on src/Makefile:
```diff
@@ -86,7 +87,7 @@ LIBPATHS=
 #	thes use the form: #include <header>
 #	source file directories are NOT auto-included here
-SYSTEM_INCLUDE_PATHS = 
+SYSTEM_INCLUDE_PATHS = /system/develop/headers/private/interface/
```
Needs /system/develop/headers/private/shared for string_for_size

humdingerb commented on src/Makefile:
```diff
@@ -149,4 +150,4 @@ DRIVER_PATH =
 DEVEL_DIRECTORY := \
 	$(shell findpaths -r "makefile_engine" B_FIND_PATH_DEVELOP_DIRECTORY)
-include $(DEVEL_DIRECTORY)/etc/makefile-engine
+include $(DEVEL_DIRECTORY)/etc/makefile-engine 
```
Won't build with the trailing space

bach5000 replied:
I always have a random trailing whitespace :/

humdingerb replied:
You can configure Pe to remove trailing whitespaces under Preferences - Files.

bach5000 replied:
Wait, really? How come I didn't know about this feature sooner :(

humdingerb commented on src/MainWindow.cpp:
```diff
@@ -12,6 +12,8 @@
 #include "SlayerApp.h"
 #include "AboutWindow.h"
+#include <AboutWindow.h>
+#include <Catalog.h>
```
Need to define B_TRANSLATE_CONTEXT

OTOH, why only do the localization for the AboutWindow?

bach5000 replied:
@humdingerb do you mean I have to use localization for all the sources???
Or do you mean I shouldn't use localization?

humdingerb replied:
Adding localization only makes sense when done for the whole app.

bach5000 replied:
@humdingerb Okay.

bach5000 commented:
Updates
* Localizations
* Fixed the signature thingy
* Got rid of SLAYER_ABOUT in Slayer.rdef
* Included <StringForSize.h>
* Added more paths to the Makefile

I tested this on gcc2 and it works fine. And finally there's a logo in the "About Slayer" window :)
Also, if I missed any strings (for localization), please don't hesitate to comment.

humdingerb commented:
It doesn't compile here. There's some garbage characters in line 700 of the MainWindow.cpp.
Even after that it doesn't build. Can you fix that?

humdingerb commented:
TeamListView seems to miss localization.
All the output with printf should not be localized. Those are only used for submitting bugreports, which gets complicated when those arrive in languages the dev doesn't understand. :)
At least, that's the current philosophy...

bach5000 commented:
Hmmm...
@humdingerb that's strange. My version of Slayer works perfectly fine. At least on gcc2.
I'll fix the localizations, and I'll try to see what might have gone wrong.

humdingerb commented:
After removing the garbage from l700 of MainWindow.cpp I get this: http://sprunge.us/EaNG

bach5000 commented:
Oh wait...
I think I have the outdated version of Slayer (I downloaded the source via outdated recipe)

bach5000 commented:
Wait a minute...
I thought I localized TeamListView????
What's going on?

humdingerb commented:
Better re-fork and clone then... :)

humdingerb commented:
Good. Back on track.
Bad news: To me it looks like starting to localize was premature. The ColumnListView and the other stuff from Santa's Gift Bag isn't layout management aware, and therefore inherently not localization-safe.
You can see that most widgets used by Slayer also use the old layout-unaware constructors. Layout-aware c'tors don't have a BRect for the frame as first parameter and don't have flags like B_FOLLOW_ALL_SIDES etc.

Probably best to remove those B_TRANSLATE macros etc.

bach5000 commented:
@humdingerb Okay :(

bach5000 commented:
@humdingerb I updated some things, but I don't have Haiku available. I'll test these changes later.

bach5000 commented:
Wait... what did I just click /:o

scottmc commented:
Check the Haiku coding guidelines on ordering of the includes.  Although not required for app on HaikuArchives, it's probably still a good idea to follow when you can:
https://www.haiku-os.org/development/coding-guidelines/

`
Alphabetize #include statements, group "includes" and \<includes\> separately.
While the header that belongs to a source file should be included first to ensure self containment, all other headers are specified from most general (POSIX) to most specific (local directory). In order to alphabetize them, please group them by API origin, like (but without the comments, of course):

#include "ThisClass.h"

// POSIX API headers
#include <stdio.h>
#include <string.h>

// Haiku API headers
#include <file.h>
#include <os.h>

#include <privateheader.h>

#include "OtherLocalHeaders.h"
`

bach5000 commented:
Will do @scottmc

scottmc commented:
@humdingerb it's showing changes requested, but I think those are from prior to the translate undo?  I'll let you make the call on merge.  If you can't merge, then just mark it ready when you have reviewed and I'll do merge later if it's ok.

humdingerb commented on src/TeamListView.cpp:
```diff
@@ -23,16 +23,16 @@ TeamListView::TeamListView(BRect frame, const char *name)
 
-#include "TeamListView.h"
 #include "CLVColumn.h"
+#include "TeamListView.h"
```
This reordering has broken the build. Now CLVColumn.h isn't aware of BView, for example. Revert this one.
Re-ordering of includes is best done in small steps with clean builds inbetween.
Do make sure to do a clean build before creating/updating a PR. The most innocuous little change has the potential to break things... :)

scottmc replied:
My bad.  I'll let bach5000 make the fix and retest it.

bach5000 replied:
Oops.

humdingerb commented:
Now that github's automatic merge check finally finished... Let's merge!
Thanks very much!

humdingerb commented:
@bach5000 : Try to use more decriptive commit messages in the future. Don't describe which files you touched - this can be seen in the commit history - but why and how something was changes to what purpose.

Also, with "git commit -a --amend" you can add your changes to the previous commit and edit the commit message. That way we don't have to squash them all into one at the end.
After amending, "git push -f" the branch to update the PR.

bach5000 commented:
@humdingerb Sorry for the late response (I should be more alert)
Okay. I'll be more descriptive in my commit messages, and I'll try to use 'git' commands instead of using the Github website ...
